collections/src/main/kotlin/ktx/collections/maps.kt

```diff
@@ -422,3 +422,63 @@ inline fun <Key, Type, Value : Iterable<Type>> GdxMap<Key, out Value>.flatten():
 inline fun <Key, Value, R> GdxMap<Key, Value>.flatMap(transform: (Entry<Key, Value>) -> Iterable<R>): GdxArray<R> {
   return this.map(transform).flatten()
 }
+
+/**
+ * Returns the value for the given [key]. If the [key] is not found in the map,
+ * calls the [defaultValue] function, puts its result into the map under the given [key] and returns it.
+ */
+inline fun <Key, Value> GdxMap<Key, Value>.getOrPut(key: Key, defaultValue: () -> Value): Value {
+  var value = this[key]
+
+  if (value == null) {
```

Comment:
This will not work correctly if LibGDX maps can store `null` values. It would be better to check if the key is present in the map. Please add a test case for this.

Reply:
Same goes for other maps.

Reply:
The reason I went for that approach is to avoid a second call to `locateKey` which should be slighty more efficient than this approach:

```Kotlin
if(key in map) {
  return map[key] // second time locateKey is called
} else {
  // ...
}
```

But, as always, you are correct that this will not work in case you put null values to the map. I did not consider that and will change it.

Reply:
> The reason I went for that approach is to avoid a second call to l`ocateKey` which should be slighty more efficient

That's what I thought. You can (kind of) avoid the second call with something along the lines of:

```kotlin
val value = map[key]
if (value == null && key !in map) {
  // ...
}
```
In best case scenario, it will perform only a single lookup and return the existing value.

Reply:
Now that I think about, what happens if you try to use a `null` key? It's OK if it throws an NPE error - you don't need to change the function itself, but maybe an extra test for the thrown exception would make sense just to document the behavior.

Reply:
How can I test that? :D Key is a not nullable type and therefore I cannot pass null?

Reply:
Kotlin type system strikes again, we might not have to test for this after all.

Anyway, you could explicitly define the key type:

```kotlin
val map = gdxMapOf<String?, String?>()
map.put(null, null) // Compiles.
```

Reply:
That is true but does that ever make sense? Why would I use a nullable key in Kotlin?

Reply:
The only reason I can think of is receiving a Java type (`Any!`) and falsely assuming is not null. You can easily avoid this is in pure Kotlin code, but interacting with Java can lead to weird errors like think.

Since type system mostly prevents you from making this error, I'll leave it up to you. If you think the extra tests could be helpful, include them in the PR. Otherwise you can resolve the discussion.

Reply:
I added test cases for it and documented the cases where a `java.lang.IllegalArgumentException` is thrown.

`IntMap` does not allow null keys and `GdxArrayMap` somehow supports null keys :D